src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -502,3 +506,90 @@ export const updateForm = (
     return transformMongoError(error)
   })
 }
+
+const replaceSettingsToUpdate = (
+  originalForm: IPopulatedForm,
+  updateKeys: (keyof SettingsUpdateDto)[],
+  body: SettingsUpdateDto,
+) => {
+  return Result.fromThrowable(
+    () => {
+      return updateKeys.reduce((accumulatedForm, key) => {
+        switch (key) {
+          // Non-object patches, assign key to form.
+          case 'authType':
+          case 'emails':
+          case 'esrvcId':
+          case 'hasCaptcha':
+          case 'inactiveMessage':
+          case 'status':
+          case 'submissionLimit':
+          case 'title': {
+            set(accumulatedForm, key, body[key])
+            break
+          }
+          // Object patches, deep mutate.
+          case 'webhook': {
+            merge(accumulatedForm[key], body[key])
+            break
+          }
+          default:
+            // Compile time check for missed keys. Thrown so unknown keys will
+            // be caught if unknown keys somehow happens during runtime.
+            // eslint-disable-next-line typesafe/no-throw-sync-func
+            throw new UnreachableCaseError(key)
+        }
```

Comment:
TypeScript wouldn't compile if the keys were not exhaustive, right?
```suggestion
```

Reply:
Typescript only complains if we specify a return type and there is no default statement, meaning the return type has a possibility to be void.

However, in this case we are mutating an object and may not return anything. Maybe I can return something in the loop instead!

```ts
updateKeys.reduce((accumulatedForm, key) => {
        switch (key) {
          // Non-object patches, assign key to form.
          case 'authType':
          case 'emails':
          case 'esrvcId':
          case 'hasCaptcha':
          case 'inactiveMessage':
          case 'status':
          case 'submissionLimit':
          case 'title': {
            set(accumulatedForm, key, body[key])
            return accumulatedForm
          }
          // Object patches, deep mutate.
          case 'webhook': {
            merge(accumulatedForm[key], body[key])
            return accumulatedForm
          }
        }
```

Now we don't need the whole throwable thing, since yea typescript won't compile! Nice catch, didn't see it then!
![image](https://user-images.githubusercontent.com/22133008/111412589-a2303d00-8717-11eb-97aa-5423fac9c990.png)

Reply:
Fixed in 5e6be06